Events app refactor

Ezi-code opened the pull request:
I was working on this PR and I found out some things I would like us to discuss first. 
@JohananOppongAmoateng @Darkbeast-glitch

Ezi-code committed:
work in progress

Ezi-code committed:
added portions for comments

Ezi-code commented on speakwise/speakwise/speakwise/events/models.py:
```diff
@@ -106,6 +106,7 @@ class Session(TimestampedModel):
         related_name="sessions",
     )
+    # location could be a model that contain country
     location = models.CharField(max_length=255, null=True)
```
Location could be a model that contains the country model and the region model, and that would make total sense. Example:
```json
{
  "id": 1,
  "name": "Main Hall",
  "address": "123 Main St",
  "city": "Lagos",
  "state": "Lagos",
  "postal_code": "100001",
  "latitude": 6.5244,
  "longitude": 3.3792,
  "description": "Main conference hall",
  "country": {
    "id": 1,
    "name": "Nigeria",
    "code": "NG"
  }
}
```

Feel-The-AGI replied:
Let's rename "name" to Venue since this is for locations

Ezi-code replied:
> Let's rename "name" to Venue since this is for locations

yh sure

Ezi-code commented on speakwise/speakwise/speakwise/events/serializers_extended.py:
```diff
@@ -5,6 +5,7 @@
 
 
+# is this really necessary? I think we can add this to the serializers file and inherit from the EventSerializer as we did here.
 class EventWithGuestSpeakersSerializer(EventSerializer):
```
Do you think this is necessary? I think we can add this to the serializers file and inherit from the EventSerializer as we did here.

Feel-The-AGI replied:
Was going to tag Julius, but it wasn't working.
After looking, i see it could benefit from inheriting it... same to avoid overhead and circular import complications later.

Ezi-code commented on speakwise/speakwise/speakwise/events/tag_models.py:
```diff
@@ -4,6 +4,7 @@
 
 
+# think this model is duplicated.
 class Tag(TimestampedModel):
```
I think this model is duplicated. There is a Tag model in `events.models`

Feel-The-AGI replied:
Looks like it's inheriting it

Ezi-code replied:
> Looks like it's inheriting it

No, it's not. Just a duplicate model

Feel-The-AGI replied:
Ok then let's kindly get rid of it as suggested

Ezi-code commented on speakwise/speakwise/speakwise/events/urls.py:
```diff
@@ -94,4 +94,6 @@
 
 
-urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+urlpatterns += static(
```
I think this could be done in the main URL config. That way, we can load all media files for all models that will need them, instead of doing it separately for apps.

Feel-The-AGI replied:
I agree reducing our overhead.

Ezi-code replied:
exactly.

Feel-The-AGI replied:
Yes, so let's reorganize it in the root file.

JohananOppongAmoateng replied:
> I think this could be done in the main URL config. That way, we can load all media files for all models that will need them, instead of doing it separately for apps.

Yes it should be

JohananOppongAmoateng commented on speakwise/speakwise/speakwise/events/serializers_extended.py:
- [ ] Why do we have this file I agree with Ezra it should be in the serializer file

Ezi-code committed:
refactoring in progress

Ezi-code committed:
update gitignore file

Ezi-code committed:
clean migration files

# Conflicts:
#	speakwise/speakwise/speakwise/organizers/models.py
#	speakwise/speakwise/speakwise/speakers/admin.py

Ezi-code committed:
removed extended serializer and tag model files

Ezi-code committed:
remove .idea env dir

Ezi-code commented:
The Event model currently looks like this:

```json
{
  "id": 0,
  "event_image": "string",
  "tags": [
    {
      "id": 0,
      "name": "string",
      "color": "string"
    }
  ],
  "website": "string",
  "short_description": "string",
  "location": {
    "id": 0,
    "country": {
      "id": 0,
      "name": "string",
      "code": "string"
    },
    "venue": "string",
    "address": "string",
    "city": "string",
    "state": "string",
    "postal_code": "string",
    "latitude": "-7.783",
    "longitude": "28.06",
    "description": "string"
  },
  "name": "string",
  "date": "string",
  "date_range": "string",
  "attendees": "string",
  "speakers": "string",
  "created_at": "2025-07-07T15:11:49.584Z",
  "updated_at": "2025-07-07T15:11:49.584Z",
  "title": "string",
  "event_nickname": "string",
  "description": "string",
  "start_date_time": "2025-07-07T15:11:49.584Z",
  "end_date_time": "2025-07-07T15:11:49.584Z",
  "is_active": true,
  "document": "string"
}
```

Ezi-code committed:
resolve errors from speakers admin

Ezi-code committed:
resolve conflicts

Ezi-code committed:
resolve migrations

Ezi-code committed:
delete .idea
